From c23333c7228656a75d4e69eac7f937a07b9b2495 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:27:11 -0400
Subject: [tools] perf/traceevent: Add flags NOHANDLE and PRINTRAW to individual events

Message-id: <1408462094-14194-263-git-send-email-jolsa@redhat.com>
Patchwork-id: 88293
O-Subject: [PATCH RHEL7.1 BZ1131394 262/325] tools lib traceevent: Add flags NOHANDLE and PRINTRAW to individual events
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit c6c2b960b7a4105f096499fba3df65d6c0272a20
Author: Steven Rostedt <srostedt@redhat.com>
Date: Fri Nov 1 17:53:59 2013 -0400

description
===========
Add the flags EVENT_FL_NOHANDLE and EVENT_FL_PRINTRAW to the event flags
to have the event either ignore the register handler or to ignore the
handler and also print the raw format respectively.

This allows a tool to force a raw format or non handle for an event.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/lib/traceevent/event-parse.c b/tools/lib/traceevent/event-parse.c
index 85cbbdd..fc6f35f 100644
--- a/tools/lib/traceevent/event-parse.c
+++ b/tools/lib/traceevent/event-parse.c
@@ -4446,11 +4446,11 @@ void pevent_event_info(struct trace_seq *s, struct event_format *event,
 {
  int print_pretty = 1;
 
- if (event->pevent->print_raw)
+ if (event->pevent->print_raw || (event->flags & EVENT_FL_PRINTRAW))
   print_event_fields(s, record->data, record->size, event);
  else {
 
-  if (event->handler)
+  if (event->handler && !(event->flags & EVENT_FL_NOHANDLE))
    print_pretty = event->handler(s, record, event,
             event->context);
 
diff --git a/tools/lib/traceevent/event-parse.h b/tools/lib/traceevent/event-parse.h
index 9ab6367..dc8539e 100644
--- a/tools/lib/traceevent/event-parse.h
+++ b/tools/lib/traceevent/event-parse.h
@@ -308,6 +308,8 @@ enum {
  EVENT_FL_ISBPRINT = 0x04,
  EVENT_FL_ISFUNCENT = 0x10,
  EVENT_FL_ISFUNCRET = 0x20,
+ EVENT_FL_NOHANDLE = 0x40,
+ EVENT_FL_PRINTRAW = 0x80,
 
  EVENT_FL_FAILED  = 0x80000000
 };
-- 
1.7.1